tree/treeplayer/inc/ROOT/TDFInterface.hxx

```diff
@@ -1099,12 +1113,14 @@ protected:
                files[slot] = merger.GetFile();
                trees[slot] = new TTree(treename.c_str(), treename.c_str());
                trees[slot]->ResetBit(kMustCleanup);
+               trees[slot]->SetAutoFlush(autoflush);
                // hack to call TTree::Branch on all variadic template arguments
                std::initializer_list<int> expander = {(trees[slot]->Branch(bnames[S].c_str(), &args), 0)..., 0};
                (void)expander; // avoid unused variable warnings for older compilers such as gcc 4.9
             }
             trees[slot]->Fill();
-            if (files[slot]->GetBytesWritten() >= cachesize) files[slot]->Write();
+            if (autoflush > 0 && trees[slot]->GetEntries() % autoflush == 0)
```

Comment:
For the record, the disconnect was me mis-reading the current code which instead of using tree->GetAutoFlush() was using the parameter passed by the user.   Using tree->GetAutoFlush is essential as the value changes as the TTree move from the first events (where GetAutoFlush is negative and expressed the desired size in compressed data size and during which  the real cluster size is being determined) to the normal regime (when GetAutoFlush returns the size of the cluster in number of event).